### ISSUE TITLE
Out Dated Code in Linear Regression with Synthetic Data Learning model

### ISSUE THREAD

**Issue by c0rRupT9** [googlecolab/colabtools#4750]
**Describe the current behavior**
There are two main code errors here due to the presence of outdated code which would run on old versions of TF:
                1) The first on is at the instance of model compilation 

  model.compile(optimizer=tf.keras.optimizers.experimental.RMSprop(learning_rate=my_learning_rate),
                loss="mean_squared_error",
                metrics=[tf.keras.metrics.RootMeanSquaredError()])

an error is thrown which is due to the incorrect reference to the RMSprop optimizer within the experimental module. In recent versions of TensorFlow and Keras, the optimizers are directly available from the tf.keras.optimizers module.

 2) The second one occurs due to that the data type passed to the training function is a list, which is not recognized by the Keras data adapters. The input data for training should be in a NumPy array, Pandas DataFrame, or a TensorFlow Dataset.

                    my_feature = ([1.0, 2.0,  3.0,  4.0,  5.0,  6.0,  7.0,  8.0,  9.0, 10.0, 11.0, 12.0])
                    my_label   = ([5.0, 8.8,  9.6, 14.2, 18.8, 19.5, 21.4, 26.8, 28.9, 32.0, 33.8, 38.2])

These may cause hindrances for beginners or new learners as they can't identify such problems please patch it as soon as possible

**Describe the expected behavior**
       The code never runs without clearing these two problems.

**What web browser you are using**
(Chrome, Firefox, Safari, etc.)

**Additional context**
Link to a minimal, public, self-contained notebook that reproduces this issue.
 - Share the file using your GitHub account using  File > Save a copy as a GitHub Gist.
 - or Share Drive notebooks using the Share button then 'Get Shareable Link'.

**Comment by metrizable** [comment]
@c0rRupT9 Thanks for filing the issue and thanks for using Colab. Please a minimal reproducible notebook which is viewable by our team.

**Comment by c0rRupT9** [comment]
Sure here's the link to the notebook https://colab.research.google.com/github/google/eng-edu/blob/main/ml/cc/exercises/linear_regression_with_synthetic_data.ipynb?hl=en#scrollTo=wDlWLbfkJtvu

**Comment by c0rRupT9** [comment]
Let me know when fixed

**Comment by sandersk** [comment]
Hi @c0rRupT9, this issue has been fixed now. You should be able to access the updated notebook at https://colab.research.google.com/github/google/eng-edu/blob/main/ml/cc/exercises/linear_regression_with_synthetic_data.ipynb.

Let us know if you encounter any further issues.

**Comment by cperry-goog** [comment]
Thank you!
